packages/http/fetch/test/common/fetchRequestAdapter.ts

```diff
@@ -124,6 +135,55 @@ describe("FetchRequestAdapter.ts", () => {
 			});
 		}
 	});
+	describe("send enum", () => {
+		for (const statusCode of [200, 201, 202, 203]) {
+			const enumResponse = "a";
+			it(`should return object for status code ${statusCode}`, async () => {
+				const mockHttpClient = new HttpClient();
+				mockHttpClient.executeFetch = async (url: string, requestInit: RequestInit, requestOptions?: Record<string, RequestOption>) => {
+					const response = new Response(enumResponse, {
+						status: statusCode,
+					} as ResponseInit);
+					response.headers.set("Content-Type", "text/plain");
+					return Promise.resolve(response);
+				};
+				const mockFactory = new TextParseNodeFactory();//new JsonParseNodeFactory();
+				const requestAdapter = new FetchRequestAdapter(new AnonymousAuthenticationProvider(), mockFactory, undefined, mockHttpClient);
+				const requestInformation = new RequestInformation();
+				requestInformation.URL = "https://www.example.com";
+				requestInformation.httpMethod = HttpMethod.GET;
+				const result = await requestAdapter.sendEnum(requestInformation, TestEnumObject, undefined);
+				assert.isDefined(result);
+				assert.equal(result, TestEnumObject.A);
```

Comment:
shouldn't we be able to use test enum instead of test enum object?

Reply:
The nodeParser at the moment expects the object itself:

``` Typescript
public getEnumValue = <T>(type: any): T | undefined => {
    const rawValue = this.getStringValue();
    if (!rawValue) {
      return undefined;
    }
    return type[toFirstCharacterUpper(rawValue)];
  };
```

I couldn't find a way to verify if a node value is of desired `enum` type without any reference to the object

Reply:
I get the fact that we need to pass the object to the request adapter method.
My comment was more about the assert line (157). I don't think we should be asking people to use the object, that's a poor developer experience. So I'm asking if we can update the unit test to work with the enum instead (for the assert part)